Manila functional tests infrastructure

sbjorn opened the pull request:
- This job introduces a heat stack with the purpose of running a Scality RING with two connectors.
- The RING installation on the stack is handled through fab tasks.
- Devstack with Manila is installed on the jclouds instance, and is configured to target the connectors (NFS and CIFS) on the heat stack.
- Manila is configured by leveraging the newly introduced https://github.com/scality/devstack-plugin-scality plugin.
- Tunnels are setup from the provder network to the connectors to allow tenants to mount shares. This depends on scality/devstack-plugin-scality#2.

Commit:
Add Manila job

Commit:
Manila functional tests: Collect artifacts

sbjorn commented on jenkins/manila-functional-tests/20-install-openstack.sh:
```diff
@@ -0,0 +1,21 @@
+cd ${SCRIPT_DIR}
+
+# Remove any deployed resources on error.
+trap "set +u && source heat-venv/bin/activate && fab destroy" ERR
```
This solution for teardown is a bit ugly. An alternative I have thought of is to handle this teardown in the short script which initiate the job on the jenkins side. This has the downside of starting jobs in different ways, which might be confusing.

JordanP replied:
I can live with how you did it now.

JordanP commented on jenkins/manila-functional-tests/20-install-openstack.sh:
```diff
@@ -0,0 +1,21 @@
+
+source /tmp/manilaci-hosts
+git clone https://github.com/openstack-dev/devstack.git
+cp devstack/samples/local.conf devstack
```
Start from a clean/empty local.conf. We don"t really know what's actually in devstack/samples/local.conf and it can change over time.

JordanP commented on jenkins/manila-functional-tests/fabfile.py:
```diff
@@ -0,0 +1,432 @@
+    )
+
+    # Add GPG key.
+    sudo('gpg --keyserver pool.sks-keyservers.net --recv-keys %s' % gpg_key)
```
Ship the public key in the repo. I.e don't rely on these key servers, they suck.

Commit:
Manila functional tests: Add apt repo key from file

JordanP commented on jenkins/manila-functional-tests/fabfile.py:
```diff
@@ -0,0 +1,432 @@
+    )
+
+    # Add GPG key.
+    put('../scality5.gpg', '/tmp')
```
It would be best to have this gpg key in that repo so that we don't depend on the file hierarchy and file names in the openstack-ci-repo.

Commit:
Manila job: Start from a blank local.conf for Devstack

This way, we know exactly what's in there instead of relying on the
provided sample conf file.

Commit:
Manila Job: work around failures when collecting artifacts

Weird failures that I can't really explain. Let's see if we can still
collect all of our artifacts.

Commit:
Manila: Configure network path to connectors

JordanP commented:
This is good enough. Let's merge it :)
